### ISSUE TITLE
Fork

### ISSUE THREAD

**Issue by pierreozoux** [MarvAmBass/docker-nginx-ssl-secure#1]
Hi,

I really liked your work, it was bothering me, how to automatically provision an instance of owncloud or piwik without playing around with this mouse..

But, unfortunately, I'm running behind an ssl offloader HAproxy, so I had to rewrite big chuncks of your code. And then, I also changed some other stuff to fit to my infra..

I have to admit, I'm sad, but I spent 3 days working on that to fir my needs. I'm afraid I can't offer a PR...

If you want to have a look, I used your work in `nginx-php`, `piwik` and `owncloud`.
https://github.com/pierreozoux/IndiePaaS/tree/master/dockerfiles/services

Among other things here are some comments:
- ENV DEBIAN_FRONTEND noninteractive
- rm -rf /var/lib/apt/lists/*
- mysql-check.sh (to check db before doing anything)
- s/chmod a+x/chmod u+x/g (try to set minimum rights)
- #!/bin/bash -eux (insteand of: [#/bin/bash](https://github.com/MarvAmBass/docker-owncloud/blob/master/startup-owncloud.sh#L1))
- /tmp/nginx.log
- support for persistent volume
- use of $URL instead of localhost

I mean the work I did is still not perfect, but it fits my needs.
I liked the way you worked, this curl idea to configure is really nice! And your images are just with minimum, not like a bunch of fat images out there.

I would love to work with you to maintain these images. I'd prefer than by maintaining them by myself.

Tell me what do you think about it?

Ah, also, I removed references to your work, as it is quiet far from that now, does it bother you? If yes, I'll put a link to your images, and a big thanks!

**Comment by MarvAmBass** [comment]
Hi pierre,

good to hear that my images could help you.

I've just took a small look above the piwik Dockerfile, seems nicly handcrafted.

I allways want to force myself to use persistent volumes instead of the host folder mapping stuff - but actually I'm always to lazy to use it. You should have a good backup/restore strategy for volume containers.

Yeah thanks on this one, it took me about 2 days to get only the minimum required parameters to set up piwik via curl. But beware, if piwik changes something we are fucked at this point, and need to update it over again.

It already costs a lot of time to maintain my own images, especially because I want to release only super stable images.

I'll keep you updated if I wanna share the maintaining work ;)

**Comment by pierreozoux** [comment]
Yeah, it takes too much time... and docker didn't solve the issue of "one container to fit all needs" far from that..
For now, I just use piwik and owncloud, just to let you know.

Cheers, and have a nice week-end!

PS: I'm curious for which project do you use these images?
PPS: would you be interested of coming to [OuiShareLabs](https://trello.com/b/J1PpskXG/ouishare-labs-camp-3)
